src/actions/walletActions.js

```diff
@@ -207,18 +214,41 @@ export function spendSlices(inputs, changeSlice) {
     );
 
     const updatedSlices = await Promise.all(sliceUpdates);
+    const firstSlice = updatedSlices[0];
+    const utxoCount = firstSlice.change
+      ? changeNodes[firstSlice.bip32Path].utxos.length
+      : depositNodes[firstSlice.bip32Path].utxos.length;
+
+    // want to make sure we only update the slices if they
+    // have changed states, i.e. the number utxos is different from what we have
+    // so if the slice state isn't reporting as changed after fetching
+    // we will try a few more times
+    if (firstSlice.utxos.length === utxoCount && retries)
+      return setTimeout(
+        () => dispatch(updateTxSlices(changeSlice, retries - 1)),
+        750
+      );
 
-    updatedSlices.forEach((slice) => {
+    // if we're out of retries and counts are still the same
+    // then we're done trying and should show an error
+    if (firstSlice.utxos.length === utxoCount && !retries) {
+      let message = `There was a problem updating the wallet. 
+It's possible your transaction did not broadcast correctly. Please confirm 
+the state of the transaction with an external block explorer, then refresh the wallet and try again.`;
```

Comment:
Perhaps we should make this sound a little less onerous?  I just saw this message because the retries were exhausted but the transaction request was a 200.  My transaction has already been confirmed so this message is completely false for me.  Wondering if we should just say something about there being a problem updating the balance as this may have absolutely nothing to do with a broadcast issue.

Reply:
Plus it would make the message shorter :smirk:

Reply:
Updated message.